.github/workflows/build.yml

```diff
@@ -120,6 +120,51 @@ jobs:
       - name: Run Tests
         run: npm run wdio-cht-form
 
+  k3de2e:
+    needs: build
+    name: k3de2e
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ env.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+        if: ${{ env.INTERNAL_CONTRIBUTOR }}
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.ECR_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.ECR_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-2
+        if: ${{ env.INTERNAL_CONTRIBUTOR }}
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        if: ${{ env.INTERNAL_CONTRIBUTOR }}
+        uses: aws-actions/amazon-ecr-login@v2
+        with:
+          mask-password: 'true'
+
+      - name: Use Node.js ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      - uses: actions/checkout@v4
+
+      - run: mkdir tests/logs
+      - run: python -m pip install git+https://github.com/medic/pyxform.git@medic-conf-1.17#egg=pyxform-medic
+      - run: npm install -g cht-conf
+      - run: npm ci
```

Comment:
I think our previous attempts of reusing code was unsuccessful, simply because GH code reuse is pretty cumbersome (like: https://docs.github.com/en/actions/creating-actions/creating-a-composite-action). 
I would love if this were just easily possible.